src/lightning_lite/plugins/precision/native_amp.py

```diff
@@ -0,0 +1,102 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from contextlib import contextmanager
+from typing import Any, Dict, Generator, Optional, Union
+
+import torch
+from torch import Tensor
+from torch.nn import Module
+from torch.optim import LBFGS, Optimizer
+
+from lightning_lite.plugins.precision.mixed import MixedPrecisionPlugin
+from lightning_lite.utilities.enums import AMPType
+from lightning_lite.utilities.imports import _TORCH_GREATER_EQUAL_1_10
+
+if _TORCH_GREATER_EQUAL_1_10:
+    from torch import autocast as new_autocast
+else:
+    from torch.cuda.amp import autocast as old_autocast
+
+
+class NativeMixedPrecisionPlugin(MixedPrecisionPlugin):
+    """Plugin for Native Mixed Precision (AMP) training with ``torch.autocast``.
+
+    Args:
+        precision: Whether to use ``torch.float16`` (``16``) or ``torch.bfloat16`` (``'bf16'``).
+        device: The device for ``torch.autocast``.
+        scaler: An optional :class:`torch.cuda.amp.GradScaler` to use.
+    """
+
+    backend = AMPType.NATIVE
+
+    def __init__(
+        self, precision: Union[str, int], device: str, scaler: Optional[torch.cuda.amp.GradScaler] = None
+    ) -> None:
+        super().__init__()
+        if precision == "bf16" and not _TORCH_GREATER_EQUAL_1_10:
+            raise ImportError("To use bfloat16 with native amp you must install torch greater or equal to 1.10.")
+        if scaler is None and precision == 16:
+            scaler = torch.cuda.amp.GradScaler()
+        if scaler is not None and precision == "bf16":
+            raise ValueError(f"`precision='bf16'` does not use a scaler, found {scaler}.")
+        self.precision = precision
+        self.device = device
+        self.scaler = scaler
+
+    @contextmanager
+    def forward_context(self) -> Generator[None, None, None]:
+        with self._autocast_context_manager():
+            yield
+
+    def backward(self, tensor: Tensor, model: Optional[Module], *args: Any, **kwargs: Any) -> None:
+        if self.scaler is not None:
+            tensor = self.scaler.scale(tensor)
+        super().backward(tensor, model, *args, **kwargs)
+
+    def optimizer_step(
+        self,
+        optimizer: Optimizer,
+        *args: Any,
+        model: Optional[Module] = None,
+        **kwargs: Any,
+    ) -> Any:
+
+        if self.scaler is None:
+            # skip scaler logic, as bfloat16 does not require scaler
+            return super().optimizer_step(optimizer, *args, model=model, **kwargs)
+        if isinstance(optimizer, LBFGS):
+            raise TypeError("Native AMP and the LBFGS optimizer are not compatible.")
+
+        self.scaler.unscale_(optimizer)
+
```

Comment:
Yes you are right. It is here because it was in the code-path for non-PL execution, but it is optional. 

From amp docs:

> If [unscale_()](https://pytorch.org/docs/stable/amp.html#torch.cuda.amp.GradScaler.unscale_) is not called explicitly, gradients will be unscaled automatically during [step()](https://pytorch.org/docs/stable/amp.html#torch.cuda.amp.GradScaler.step).